cli/src/main.rs

```diff
@@ -1,16 +1,108 @@
+use clap::Parser;
 use core::eval;
+use notify::{
+    Config, PollWatcher, RecommendedWatcher, RecursiveMode, Result, Watcher, WatcherKind,
+};
 use parser::parse;
-use std::{env, fs};
+use std::env;
+use std::{fs, path::Path};
+use std::{fs::File, io::Write, time::Duration};
+use termion::{color, style};
 
-// We would likely want to use "clap" and its derive feature to describe
-// the command-line interface with a declarative struct
+#[derive(Parser, Debug)]
+#[command(author, version, about)]
+struct Args {
+    #[arg(index = 1, help = "Path to input file")]
+    input: String,
 
-fn main() {
-    let path: String = env::args().skip(1).take(1).collect();
-    let source = fs::read_to_string(path).expect("Failed to read file");
+    #[arg(index = 2, help = "Path to output file")]
+    output: String,
 
+    #[arg(
+        short = 'w',
+        long = "watch",
+        help = "Watches file and compiles changes"
+    )]
+    w: bool,
+}
+
+fn compile_file(args: &Args) -> Result<()> {
+    let source = fs::read_to_string(&args.input).expect("[ERROR] Failed to read file...");
```

Comment:
Might be nicer to print to stderr this rather than panic:ing to get a prettier error message.

Perhaps we could use  `let... if` like this
```rs
let Ok(source) = fs::read_to_string(....) else {
   eprintln!("[ERRRO] ....");
};
```

Reply:
same thing with returning error here 👍